terminal.go

```diff
@@ -597,6 +652,17 @@ func (g *LightningTerminal) startMainWebServer() error {
 			return
 		}
 
+		// REST requests aren't that easy to identify, we have to look
+		// at the URL itself. If this is a REST request, we give it
+		// directly to our REST handler which will then forward it to
+		// us again but converted to a gRPC request.
+		if g.cfg.EnableREST && isRESTRequest(req) {
+			log.Infof("Handling REST request: %s", req.URL.Path)
+			g.restHandler.ServeHTTP(resp, req)
```

Comment:
Exactly. It shouldn't be scary, this is how the REST proxy in `lnd` is working as well. Except that there we have different ports. But if you can identify what a request is (which fortunately is very easy with gRPC and grpc-web as they have their own `Content-Type` values), you can run everything on the same port.

Reply:
I would say it is quite different, since if the proxy forwards it to a different port, at least it stops there.

Here, say if a request was made that even after converted to grpc would look like a REST request it would be recursively forwarded again. I don't know enough about the conversion to know if this would be possible, it probably would error out at some point.

Reply:
Ah, I see what you mean. Once the call is given to the REST proxy/converter, it will only convert it to gRPC if it's a known request pattern (let's say `/v1/getinfo`). If it's not a registered (e.g. not in the `rest-annotations.yaml` of any of the daemons), a 404 will be returned by the proxy and the request ends. If it is known, the forwarded URI is completely different (e.g. `/lnrpc.Lightning/GetInfo`) and wouldn't match to REST again. And because for gRPC you always have `/package.ServiceName/RPCName`, we won't create an accidental match as long as we prefix all REST URIs with `/vX/`.